sled-agent/src/services.rs

```diff
@@ -3659,20 +3664,29 @@ impl ServiceManager {
                 log,
                 "Failed to take bundle of unexpected zone";
                 "zone_name" => &expected_zone_name,
-                "reason" => ?e,
+                InlineErrorChain::new(&e),
             );
         }
         if let Err(e) = zone.runtime.stop().await {
             error!(log, "Failed to stop zone {}: {e}", zone.name());
         }
+        if let Err(e) =
+            self.clean_up_after_zone_shutdown(&zone.config.zone).await
+        {
+            error!(
+                log,
+                "Failed to clean up after stopping zone {}", zone.name();
+                InlineErrorChain::new(&e),
+            );
```

Comment:
This makes me nervous: if we fail to clean up, we never try again. But that's already true for everything else this function does; e.g., if we fail to stop the zone, we also never try again. I'm not sure if I'm making this demonstrably worse. I'm also not sure what other option we have prior to moving to a zones reconciler (which I'm still marching towards!).